components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java

```diff
@@ -181,12 +182,25 @@
                 }
                 scopeSet.add(scope);
             }
-            return scopeSet;
+            return sortScopes(scopeSet);
         } else {
-            return getScopesFromExtensions(openAPI);
+            return sortScopes(getScopesFromExtensions(openAPI));
         }
     }
 
+    /**
+     * Sort scopes by name.
+     * This method was added to display scopes in publisher in a sorted manner.
+     *
+     * @param scopeSet
+     * @return Scope set
+     */
+    private Set<Scope> sortScopes(Set<Scope> scopeSet) {
```

Comment:
better move to a utility class as same method is duplicated.

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8244